dotCMS/src/main/java/com/dotcms/security/secret/SecretsStore.java

```diff
@@ -0,0 +1,70 @@
+package com.dotcms.security.secret;
+
+import java.util.List;
+import java.util.Optional;
+import com.dotmarketing.util.Config;
+import io.vavr.control.Try;
+
+public interface SecretsStore {
+
+    /**
+     * Optionally returns the secrets value if there is one, empty if not
+     * 
+     * @param variableKey
+     * @return
+     */
+    Optional<char[]> getValue(final String variableKey);
+
+    /**
+     * Lists the keys - keys can be case insensitive (forced lowercase) based on the implementation
+     * 
+     * @return
+     */
+    List<String> listKeys();
+
+    /**
+     * deletes all the entries in the secretsStore
+     * 
+     * @return
+     * @throws Exception
+     */
+    boolean deleteAll() throws Exception;
+
+    /**
+     * saves a single key and a value in the SecretsStore
+     * 
+     * @return
+     * @throws Exception
+     */
+    boolean saveValue(String variableKey, char[] variableValue);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)